src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java

```diff
@@ -391,37 +399,33 @@ private static String removeCheckstylePrefixIfExists(String checkName) {
         return values;
     }
 
-    /**
-     * Checks that annotation is empty.
-     *
-     * @param values list of values in the annotation
-     * @return whether annotation is empty or contains some values
-     */
-    private static boolean isAnnotationEmpty(List<String> values) {
-        return values == null;
-    }
-
     /**
      * Get target of annotation.
      *
      * @param ast the AST node to get the child of
      * @return get target of annotation
```

Comment:
I think it would be good to extend the javadoc in this case to explain why we use Optional (maybe mention your comment below), and extend return statement to  `a DetailAST wrapped in an {@link Optional}` or something similar.

Reply:
`a DetailAST wrapped in an {@link Optional}`
I do not see any need for it. By default if method returns Optional, it means that there can be no result (null).

May be it makes sense to add some explanation in what cases there are no target for annotation.